crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -300,106 +300,79 @@ pub struct EncryptionInfo {
+    /// This is a convenience constructor for a plaintext event when you don't
+    /// need to set `push_action`, for example inside a test.
     pub fn new(event: Raw<AnySyncTimelineEvent>) -> Self {
-        Self {
-            inner_event: event,
-            inner_encryption_info: None,
-            push_actions: vec![],
-            unsigned_encryption_info: None,
-        }
+        Self { inner: TimelineEventInner::PlainText { event }, push_actions: vec![] }
     }
 
     /// Create a new `SyncTimelineEvent` from the given raw event and push
     /// actions.
     ///
-    /// This is a convenience constructor for when you don't need to set
-    /// `encryption_info`, for example inside a test.
+    /// This is a convenience constructor for a plaintext event, for example
+    /// inside a test.
     pub fn new_with_push_actions(
         event: Raw<AnySyncTimelineEvent>,
         push_actions: Vec<Action>,
     ) -> Self {
-        Self {
-            inner_event: event,
-            inner_encryption_info: None,
-            push_actions,
-            unsigned_encryption_info: None,
-        }
+        Self { inner: TimelineEventInner::PlainText { event }, push_actions }
     }
 
     /// Get the event id of this `SyncTimelineEvent` if the event has any valid
     /// id.
     pub fn event_id(&self) -> Option<OwnedEventId> {
-        self.inner_event.get_field::<OwnedEventId>("event_id").ok().flatten()
+        self.inner.event().get_field::<OwnedEventId>("event_id").ok().flatten()
     }
 
     /// Returns a reference to the (potentially decrypted) Matrix event inside
     /// this `TimelineEvent`.
     pub fn event(&self) -> &Raw<AnySyncTimelineEvent> {
-        &self.inner_event
+        self.inner.event()
     }
 
     /// If the event was a decrypted event that was successfully decrypted, get
     /// its encryption info. Otherwise, `None`.
     pub fn encryption_info(&self) -> Option<&EncryptionInfo> {
-        self.inner_encryption_info.as_ref()
+        self.inner.encryption_info()
     }
 
     /// Takes ownership of this `TimelineEvent`, returning the (potentially
     /// decrypted) Matrix event within.
     pub fn into_event(self) -> Raw<AnySyncTimelineEvent> {
-        self.inner_event
+        self.inner.into_event()
     }
 
     /// Replace the Matrix event within this event. Used to handle redaction.
     pub fn replace_inner(&mut self, event: Raw<AnySyncTimelineEvent>) {
-        self.inner_event = event;
+        self.inner = TimelineEventInner::PlainText { event };
     }
 }
 
 #[cfg(not(tarpaulin_include))]
 impl fmt::Debug for SyncTimelineEvent {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let SyncTimelineEvent {
-            inner_event,
-            inner_encryption_info,
-            push_actions,
-            unsigned_encryption_info,
-        } = self;
+        let SyncTimelineEvent { inner, push_actions } = self;
```

Comment:
Maybe this custom `Debug` ain't needed anymore, and we can derive Debug on `SyncTimelineEvent` now instead?

Reply:
No, the custom Debug implementation was there for adding the debug wrapper for the event, I don't think anyone cared about the custom formatting for `push_actions` that much.

Reply:
6e40211af removes the explicit `Debug` impl